offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/components/affiliation/update/UpdateAffiliationView.groovy

```diff
@@ -132,4 +144,82 @@ class UpdateAffiliationView extends FormLayout implements Resettable, Updatable<
       }
     }
   }
+
+  /**
+   * Modal window which prompts the user to confirm that the affiliation category change was intended
+   * @param Affiliation affiliation the to be updated affiliation for which the category was changed
+   */
+
+  private class ConfirmAffiliationChangeWindow extends Window {
+    private Button abortButton
+    private Button submitButton
+    private Affiliation updatedAffiliation
+
+    ConfirmAffiliationChangeWindow(Affiliation affiliation) {
+      this.updatedAffiliation = affiliation
+      generateAffiliationChangeWindow()
+    }
+
+    private void generateAffiliationChangeWindow() {
+      this.setCaption("Confirm affiliation update")
+      this.setContent(generateWindowContent())
+      setupWindowStyle()
+      registerListeners()
+    }
+
+    private VerticalLayout generateWindowContent() {
+      VerticalLayout windowContent = new VerticalLayout()
+      windowContent.addComponents(generateText(), generateButtonRow())
+      return windowContent
+    }
+
+    private Label generateText() {
+      Label windowText = new Label("Changing the affiliation category affects the pricing of <b> ALL </b> offers associated with this affiliation!")
+      windowText.setContentMode(ContentMode.HTML)
+      return windowText
+    }
+
+    private void setupButtons() {
+      this.abortButton = new Button("Abort Affiliation Update")
+      this.abortButton.setIcon(VaadinIcons.CLOSE_CIRCLE)
+      this.abortButton.addStyleName(ValoTheme.BUTTON_DANGER)
+      this.submitButton = new Button("Confirm Affiliation Update")
+      this.submitButton.setIcon(VaadinIcons.OFFICE)
+      this.submitButton.addStyleName(ValoTheme.BUTTON_FRIENDLY)
+    }
+
+    private HorizontalLayout generateButtonRow() {
+      setupButtons()
+      HorizontalLayout buttonLayout = new HorizontalLayout(abortButton, submitButton)
+      HorizontalLayout buttonRow = new HorizontalLayout()
+      buttonRow.setDefaultComponentAlignment(Alignment.BOTTOM_CENTER)
+      buttonRow.addComponent(buttonLayout)
+      buttonRow.setWidthFull()
+      buttonRow.setSpacing(false)
+      buttonRow.setMargin(false)
+      return buttonRow
+    }
+
+    private void setupWindowStyle() {
+      this.setWidthUndefined()
+      this.center()
+      this.setClosable(false)
+      this.setModal(true)
+      this.setResizable(false)
+    }
+
+    private void registerListeners() {
+      this.submitButton.addClickListener(withHandledException(it -> onSubmit()))
+      this.abortButton.addClickListener(withHandledException(it -> onAbort()))
+    }
+
+    private void onSubmit() {
+      triggerAffiliationUpdate(updatedAffiliation)
+      this.close()
+    }
+
+    private void onAbort() {
```

Comment:
I like that you extracted the behaviour in separate methods :+1: makes it way easier to read.

Reply:
🧠